runtime/interpreter/decode.go

```diff
@@ -1724,15 +1724,12 @@ func (d LocationDecoder) decodeAddressLocation() (common.Location, error) {
 		return nil, err
 	}
 
+	common.UseMemory(d.memoryGauge, common.NewBytesMemoryUsage(len(encodedAddress)))
```

Comment:
`decodeAddressBytes` already does this and the length check. So maybe use that.
Also can remove the TODO above (L#1695)